contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -0,0 +1,61 @@
+
+When we add a feature flag, we basically add if/else conditions to ensure that a feature is ONLY activated when either the default is on or if the deployer has switched it on explicitly.
+
+## Fields in the specification are
+	// Default is the default enablement state for the feature. possible values are true/false.
+	Default bool
+	
+    // LockToDefault indicates that the feature is locked to its default and cannot be changed. possible values are true/false.
+	LockToDefault bool
+	
+    // PreRelease indicates the maturity level of the feature. possible values are "featuregate.Alpha", "featuregate.Beta", "featuregate.GA", "featuregate.Deprecated"
+	PreRelease prerelease
+
+
+## Alpha Features
+
+* `PreRelease` is set to `featuregate.Alpha`
+* `Default` is always set to `false`
+* `LockToDefault` is not set. Defaults to `false`
+
+This enables folks to switch on the feature using the command line. By default it is not switched on.
+
+## Beta Features
+
+* `PreRelease` is set to `featuregate.Beta`
+* `Default` is set to `true`
+* `LockToDefault` is not set. Defaults to `false`
+
+This enables the feature to be on and available in the default installation of kubernetes. 
+
+Sometimes (rarely) the `Default` is set to `false`. This tells folks that while this feature is in Beta, they will still need to do some work to switch it on and use it and potentially take some other explicit action outside of Kubernetes. For example, see the [CSIMigration feature gates](https://github.com/kubernetes/kubernetes/blob/5b0a2c3a29f6b5392e0f8f94ba5669bdc9eb73f6/pkg/features/kube_features.go#L792).
+
+## GA Features
+
+* `PreRelease` is set to `featuregate.GA`
+* `Default` is always set to `true`
+* `LockToDefault` is set to `true`
+
+GA features are always on. 
+
+Sometimes (rarely) we do not set `LockToDefault` and let it default to `false`. This enables folks to switch off the GA feature. We do this to tell folks that while the feature (for example `coredns`) is GA, they need to move off say `kubedns` to `coredns` in their infrastructures pretty soon and if they want to continue `kubedns` for a short time, they will have to switch off the GA flag. When we do remove the support for `kubedns` entirely we would set `LockToDefault` to `true` with some grace period for the transition.
+
+At some time post-GA, the feature will become a part of the product and all feature gate references will be removed. Typically, we add a comment in [kubefeatures.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go) such as: `// remove in 1.23` to signal when we plan to remove the feature gate. Remember when the feature gate is removed and the deployer has forgotten to drop the reference to the feature in the CLI flags (say the `kube-apiserver`), then they will see a hard failure. 
```

Comment:
Suggest saying "two releases" (per the deprecation policy). Also, it is part of the product at GA, not post-GA... I'd probably drop that bit:

> Two releases post-GA and deprecation, the feature gate is removed.

Reply:
"At least two releases" ?